net-p2p/bitcoin-core/bitcoin-core-25.0.ebuild

```diff
@@ -0,0 +1,251 @@
+		$(use_with berkdb bdb)
+		$(use_enable systemtap usdt)
+		$(use_with upnp miniupnpc)
+		$(use_with nat-pmp natpmp)
+		$(use_enable test tests)
+		--disable-bench
+		--disable-fuzz{,-binary}
+		$(use_with qrcode qrencode)
+		--disable-ccache
+		$(use_enable asm)
+		$(use_enable zeromq zmq)
+		$(use_enable man)
+		$(use_enable external-signer)
+		--with-utils
+		$(use_enable {bitcoin,util}-cli)
+		--enable-util-tx
+		--${wallet}-util-wallet
+		--disable-util-util
+		$(use_with libs)
+		$(use_with daemon)
+		$(use_with qt5 gui qt5)
+		$(use_with dbus qtdbus)
+		$(use_with system-leveldb)
+		$(use_with system-libsecp256k1)
+	)
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	use external-signer && DOCS+=( doc/external-signer.md )
+	use berkdb || use sqlite && DOCS+=( doc/managing-wallets.md )
+	use libs && DOCS+=( doc/shared-libraries.md )
+	use systemtap && DOCS+=( doc/tracing.md )
+	use zeromq && DOCS+=( doc/zmq.md )
+
+	default
+
+	! use libs || find "${ED}" -name '*.la' -delete || die
+	! use test || rm -f -- "${ED}"/usr/bin/test_bitcoin{,-qt} || die
+
+	newbashcomp contrib/completions/bash/bitcoin-tx.bash-completion bitcoin-tx
+	use bitcoin-cli && newbashcomp contrib/completions/bash/bitcoin-cli.bash-completion bitcoin-cli
+	use daemon || use qt5 && newbashcomp contrib/completions/bash/bitcoind.bash-completion bitcoind
+
+	if use daemon ; then
+		insinto /etc/bitcoin
+		sed -ne 's/ To use, copy this file$//p;Tp;:0;n;/save the file\.$/!b0;d;:p;p' \
+			share/examples/bitcoin.conf >"${ED}/etc/bitcoin/bitcoin.conf" || die
+		fowners bitcoin:bitcoin /etc/bitcoin/bitcoin.conf
+		fperms 600 /etc/bitcoin/bitcoin.conf
+
+		newconfd contrib/init/bitcoind.openrcconf bitcoind
+		newinitd "${FILESDIR}/bitcoind.openrc" bitcoind
+		systemd_newunit contrib/init/bitcoind.service bitcoind.service
+
+		keepdir /var/lib/bitcoind
+		fperms 700 /var/lib/bitcoind
+		fowners bitcoin:bitcoin /var/lib/bitcoind
+		dosym -r /etc/bitcoin/bitcoin.conf /var/lib/bitcoind/bitcoin.conf
+		dosym -r /var/lib/bitcoind /var/lib/bitcoin/.bitcoin
+
+		doman "${FILESDIR}/bitcoin.conf.5"
+
+		insinto /etc/logrotate.d
+		newins "${FILESDIR}/bitcoind.logrotate-r1" bitcoind
+	fi
+
+	if use qt5 ; then
+		insinto /usr/share/icons/hicolor/scalable/apps
+		newins src/qt/res/src/bitcoin.svg bitcoin128.svg
+
+		domenu "${FILESDIR}/org.bitcoin.bitcoin-qt.desktop"
+
+		if use kde ; then
+			insinto /usr/share/kservices5
+			doins "${FILESDIR}/bitcoin-qt.protocol"
+			dosym ../../kservices5/bitcoin-qt.protocol /usr/share/kde4/services/bitcoin-qt.protocol
+		fi
+	fi
+
+	if use examples ; then
+		docinto examples
+		dodoc -r contrib/{linearize,qos}
+		use zeromq && dodoc -r contrib/zmq
+	fi
+}
+
+pkg_preinst() {
+	if use daemon ; then
+		if [[ ! -d "${EROOT}/var/lib/bitcoin/.bitcoin" ]] ; then
+			rm -r -- "${ED}/var/lib/bitcoin" || die
+		elif [[ ! -e "${EROOT}/var/lib/bitcoind" ]] ; then
+			elog "Moving your ${EPREFIX}/var/lib/bitcoin/.bitcoin to ${EPREFIX}/var/lib/bitcoind."
```

Comment:
I very much like that, as I never was happy with bitcoind data files in a hidden directory. I made something similar when transitioning ejabberd's home directory. Can't say that it went absolutely smooth though. Not that I am aware of a better alternative (besides not doing it at all)>

Reply:
Can you recall any gotchas that arose in that migration? I tested every path I could think of, both with source builds and binary package installations, and this did the right thing in every case, but maybe there's a scenario I didn't think of?

Reply:
It did occur to me that someone might have `/var/lib/bitcoin/.bitcoin` being a mountpoint, which would preclude its being renamed, but I assumed that to be so unlikely as not to be worth guarding against.

Reply:
I went ahead and did it. Now the migration is only performed automatically if `${EROOT}/var/lib/bitcoin/.bitcoin` is a directory (or a symlink to a directory) **and** `${EROOT}/var/lib/bitcoind` does not exist **and** `${EROOT}/var/lib/bitcoin/.bitcoin` is on the same mountpoint as `${EROOT}/var/lib`. If the mointpoints differ, then we `ewarn` about not performing the migration, delete our convenience symlink, move our keepfile to the old location, and set `BITCOIND_DATADIR` in `${ED}/etc/conf.d/bitcoind` to the old location.